### PR TITLE
5093-aggregation rest api permission issue

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Locally create a composite resource. Create a model instance aggregation.
2. Access the UI for Hydroshare REST API at /hsapi
3. Use this endpoint (/resource/{resource_id}/modelinstance/meta/{aggregation_path}) to retrieve metadata for the model instance aggregation you created in step 1.
4. Verify that you are as an owner of the resource able to retrieve aggregation metadata for this private resource.
5. Logout. Use the same endpoint (as in step 3) to access the metadata for the model instance aggregation. Verify that you are not able to retrieve metadata when the resource is private.
6. Login and make the resource discoverable.
7. Logout. Use the same endpoint (as in step 3) to access the metadata for the model instance aggregation. Verify that you are able to retrieve metadata when the resource is discoverable.
8. Login and make the resource public.
9. Logout. Use the same endpoint (as in step 3) to access the metadata for the model instance aggregation. Verify that you are able to retrieve metadata when the resource is public.

### CODE REVIEW

**Comment by devincowan** [comment]
wow this was a more in-depth fix than I expected!

**Comment by pkdash** [comment]
> wow this was a more in-depth fix than I expected! tests!

@devincowan It took more time to write the missing tests. The code change to fix the resource access permission is minimal.

**Review by devincowan — APPROVED** [review]
Nice tests Pabitra, looks good.

File: hs_core/tests/api/rest/test_model_aggregations.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from django.conf import settings
 from django.core.files.uploadedfile import UploadedFile
 from rest_framework import status
+from rest_framework.test import APIClient
```
[the remark this answers is not in the document]
Comment (pkdash) [reply hs_core/tests/api/rest/test_model_aggregations.py:9]: @devincowan I was not aware of the rest framework APIClient issue when upgrading hydroshare Django version to 3.2. Looks like you are seeing some issues as in #5099. I am assuming we can resolve that as part of #5099.

[the remark this answers is not in the document]
Comment (devincowan) [reply hs_core/tests/api/rest/test_model_aggregations.py:9]: sounds like a good plan

**Pull request merged**